### PR TITLE
Update promises with current fee before settling

### LINKED ISSUE #2893: Invalid fee when settling old promises
The stored hermes promises can not be settled via transactor due to the fact that the fees are now dynamic.

Old promise with a fee that's less than the current transaction fee can not be settled.

### PR DESCRIPTION
**Pull request by vkuznecovas**
Closes #2893

### CODE REVIEW

File: identity/registry/transactor.go [hunk 1]
```diff
@@ -77,7 +77,7 @@ type FeesResponse struct {
 
 // IsValid returns false if the fee has already expired and should be re-requested
 func (fr FeesResponse) IsValid() bool {
-	return time.Now().After(fr.ValidUntil)
+	return time.Now().UTC().After(fr.ValidUntil.UTC())
```
Comment (tomasmik) [inline identity/registry/transactor.go:80]: Am I just confused or should actually be `Before` and not after?

File: session/pingpong/hermes_promise_settler.go [hunk 2]
```diff
@@ -386,8 +393,24 @@ func (aps *hermesPromiseSettler) SettleWithBeneficiary(chainID int64, providerID
...
+	fees, err := aps.transactor.FetchSettleFees(promise.ChainID)
+	if err != nil {
+		return crypto.Promise{}, fmt.Errorf("could not fetch settle fees: %w", err)
+	}
+
+	updatedPromise, err := aps.hermesCaller.UpdatePromiseFee(promise, fees.Fee)
+	if err != nil {
+		return crypto.Promise{}, fmt.Errorf("could not update promise fee: %w", err)
+	}
+	updatedPromise.R = promise.R
```
Comment (mdomasevicius) [inline session/pingpong/hermes_promise_settler.go:407]: I am just curious - what is this `.R` ?

**Review by mdomasevicius — APPROVED** [review]

File: session/pingpong/hermes_promise_settler.go (hunk 2 above)
Comment (vkuznecovas) [reply session/pingpong/hermes_promise_settler.go:407]: R is a random number. Hashlock is derived from it.

File: identity/registry/transactor.go (hunk 1 above)
Comment (vkuznecovas) [reply identity/registry/transactor.go:80]: That might explain why it wasn't working ;)

**Review by soffokl — APPROVED** [review]

**Pull request merged**